cmd/handler-utils.go

```diff
@@ -385,6 +386,23 @@ func notFoundHandler(w http.ResponseWriter, r *http.Request) {
 	writeErrorResponse(context.Background(), w, errorCodes.ToAPIErr(ErrMethodNotAllowed), r.URL, guessIsBrowserReq(r))
 }
 
+// If the API version does not match with current version.
+func versionMismatchHandler(w http.ResponseWriter, r *http.Request) {
+	path := r.URL.Path
+	var err error
+	switch {
+	case strings.HasPrefix(path, minioReservedBucketPath+"/peer/"):
+		err = fmt.Errorf("incorrect peerRESTVersion")
+	case strings.HasPrefix(path, minioReservedBucketPath+"/storage/"):
+		err = fmt.Errorf("incorrect storageRESTVersion")
+	case strings.HasPrefix(path, minioReservedBucketPath+"/lock/"):
+		err = fmt.Errorf("incorrect lockRESTVersion")
+	default:
+		err = fmt.Errorf("incorrect adminAPIVersion")
+	}
+	writeCustomErrorResponseJSON(context.Background(), w, errorCodes.ToAPIErr(ErrNoSuchVersion), err.Error(), r.URL)
```

Comment:
I don't think everyone talks JSON it is perhaps better just to do JSON with admin API rest should be just HTTP status messages? 

Also, let's grab the entire r.URL.String() to show what is unexpected here? send this back in errorResponseJOSN

Reply:
also, don't reuse S3 errors such as ErrNoSuchVersion - define a new one specifically for AdminErr

Reply:
> I don't think everyone talks JSON it is perhaps better just to do JSON with admin API rest should be just HTTP status messages?
> 
> Also, let's grab the entire r.URL.String() to show what is unexpected here? send this back in errorResponseJOSN

You mean writing back the headers only. It will have less info right. The aim of this PR is to provide more details right

Reply:
It will fail because no other API handles json other than admin. So you can't use that as a common error for all APIs. You should write what is appropriate for each API.

Reply:
We are already tagging the `req.URL.Path` in the resource. Do you recommend the full path instead?

Reply:
Yeah full path including the query params.